### ISSUE TITLE
Run Dependancy only once

### ISSUE THREAD

**Issue by sadlil** [go-task/task#96]
For a task file like following,
```
version: '2'
tasks:
  default:
    deps:
    - fmt
    - install

  fmt:
    cmds:
    - goimports -w *.go pkg
    - gofmt -s -w *.go pkg

  install:
    deps:
    - fmt
    cmds:
    - go install . ./pkg/...

```

if I run `task` - for runnning the default task it will run the fmt twice. But running twice is not necessary. Add an option to run only once would be better.

**Comment by JulienBreux** [comment]
@sadlil just a question. 😊 

Why:
```
tasks:
  default:
    deps:
    - fmt <<<
```

**Comment by smyrman** [comment]
Duplicate of #53

**Comment by sadlil** [comment]
@JulienBreux why not? this is just a small example i gave just an example to illustrate, we could make something much bigger in real use cases, and then you do need this.
I still didn't check for a case where a circular dependency could exists or a dependency execution order though.

@smyrman sorry for the dup, failed to check.

**Comment by smyrman** [comment]
No problem @sadlil, let's continue the discussion on #53.
